src/api.rs

```diff
@@ -450,18 +450,14 @@ impl Context {
     cmp::min(next_detected.unwrap(), next_limit)
   }
 
-  fn set_frame_properties(&mut self, idx: u64) -> Result<(), ()> {
-    let props = self.get_frame_properties(idx);
-    let result = props.as_ref().map(|_| ()).map_err(|_| ());
-    match props {
-      Ok(props) | Err(props) => {
-        self.frame_data.insert(idx, props);
-      }
-    }
-    result
+  fn set_frame_properties(&mut self, idx: u64) -> bool {
+    let (fi, success) = self.get_frame_properties(idx);
+    self.frame_data.insert(idx, fi);
+
+    success
```

Comment:
Naming this "success" is weird as it really implies it's error handling and that the existing code would make more sense. Isn't this more like an EOF flag?

Reply:
There are situations other than EOF that could cause this flag to be false, mostly around frame reordering. Given that, I'm not sure what a better name for this variable might be.

Reply:
We decided for `end_of_subgop`, next set will be renaming the functions to something a bit more descriptive.